src/mds/Server.cc

```diff
@@ -3206,6 +3206,23 @@ bool Server::check_fragment_space(MDRequestRef &mdr, CDir *in)
   return true;
 }
 
+/**
+ * check whether max_entries_per_dir has reached maximum size
+ *
+ */
+bool Server::check_max_entries_per_dir(MDRequestRef &mdr, CDir *in)
+{
+  const auto size = in->inode->get_projected_inode()->dirstat.nfiles +
+                   in->inode->get_projected_inode()->dirstat.nsubdirs;
+  if (g_conf()->mds_max_entries_per_dir && size >= g_conf()->mds_max_entries_per_dir) {
```

Comment:
Cache this config like we do for e.g. `mds_max_snaps_per_dir`.

Reply:
The config number is cached now for performance.